jsk_maps: add publish_scene.l

k-okada opened the pull request:
Add node to publish building model as a Marker

1) start launch file
```
 roslaunch jsk_maps start_map_eng2.launch use_machine:=true
```
2) start publish_scene.l node
```
rosrun jsk_maps publish_scene.l
```
3) start rviz
```
rviz -d $(rospack find jsk_maps)/launch/map_eng2.rviz
```

![eng2 rviz_ - RViz (Ubuntu-18 04) 2022_07_05 0_55_27](https://user-images.githubusercontent.com/493276/177190126-5284ba9d-1cf3-4236-851d-bac57f3aa67b.png)

If we merge https://github.com/jsk-ros-pkg/jsk_roseus/pull/713, we can cleanup `publish_scene.l` node

Question :
 Do we use `start_map_eng2.launch` in pr2/fetch startup? if so , can we add `publish_scene.l` to `start_map_eng2.launch`?

Cc: @tkmtnt7000

tkmtnt7000 commented:
In pr2 and fetch, we use `start_map_eng2.launch` as default.
https://github.com/jsk-ros-pkg/jsk_robot/blob/242629882207707fc81d2b2803bfea183b6fa5aa/jsk_fetch_robot/jsk_fetch_startup/launch/fetch_bringup.launch#L177-L182
https://github.com/jsk-ros-pkg/jsk_robot/blob/a01da425852f0a407e54fe50ed59759515f2da3a/jsk_pr2_robot/jsk_pr2_startup/pr2.launch#L62-L66
I think it's ok to add `publish_scene.l` to `start_map_eng2.launch` with a boolean arg something like `use_publish_scene`

k-okada commented:
updated to visualize  transparent color
![Screenshot from 2022-07-05 17-16-12](https://user-images.githubusercontent.com/493276/177282653-7cb39a95-2950-4d2f-9508-38454fda9b51.png)

@tkmtnt7000  Please set transparent color to get better visualization, 
- elevator model and colors https://github.com/jsk-ros-pkg/jsk_demos/blob/244be9df1bc7b3c711900ce6adb546208e31b207/jsk_maps/src/eng2-scene.l#L330
- I also changed floor color at https://github.com/jsk-ros-pkg/jsk_demos/pull/1361/files#diff-577727711d6a750476a09854716e134bf638287f12e7e4ee027301433fad1eb2R49, because if we change color at https://github.com/jsk-ros-pkg/jsk_demos/blob/244be9df1bc7b3c711900ce6adb546208e31b207/jsk_maps/src/eng2-scene.l#L351, it had bad EusLisp visualization.
- If you want to set transparent color to room walls. may be you should change them within `publish_scene.l`
- you can change floor and empty rooms at https://github.com/jsk-ros-pkg/jsk_demos/blob/master/jsk_maps/src/eng2-rooms.l#L26 and https://github.com/jsk-ros-pkg/jsk_demos/blob/master/jsk_maps/src/eng2-rooms.l#L44

> I think it's ok to add publish_scene.l to start_map_eng2.launch with a boolean arg something like use_publish_scene

Please create PR @tkmtnt7000

tkmtnt7000 commented:
Thank you very much for improvement. We can now visualize eng2_building better with transparent color.

>Please create PR 

I made PR https://github.com/k-okada/jsk_demos/pull/115

tkmtnt7000 commented:
このPRを作っていただいた経緯がチャットベースだったので，記録と後の検索性のためにこちらにその内容を貼っておきます．

> Naoto Tsukamoto, 月 22:02
> @ Naoki Hiraoka 
> Rvizで73B2のeusモデルを出すプログラムがあると聞いたのですが，ご存知でしょうか？
> 
> Naoki Hiraoka, 月 22:05
> 心当たりはありませんが、恐らくhttps://github.com/jsk-ros-pkg/jsk_roseus/blob/dd8d2c5c40da926ca20129ba857a9504147cc5af/roseus/euslisp/roseus-utils.l#L461 を使うのだろうと思います。
> 
> Naoto Tsukamoto, 月 22:08
> ありがとうございます！参考にします．
> 
> Shun Hasegawa, 月 22:10
> これだったりする？
> https://github.com/jsk-ros-pkg/jsk_model_tools/tree/master/eusurdf
> 
> Yoshiki Obinata, 月 22:14
> 横からすいません，Visualize gazebo scene to RViz というものがあるのですね．ということはGazeboにJSKの部屋をspawnさせて，そこからコンバートするということでしょうか？
> 
> Naoki Hiraoka, 月 22:20
> 多分gazeboは必須ではなくて、room73b2のurdfをrosparamにあげて、joint_state_publisherとrobot_state_publisherでtf を出して、rvizのRobotModelで表示しているように見えます。
> 
> Yoshiki Obinata, 月 22:21
> なるほど，理解しました
> 
> 岡田慧, 昨日 1:23
> Eusモデルに存在するものはすべてRvizで表示できる、というのは知っているとよいです。つまり、irtviewerはもう必要なくて、rvizだけでよい、という世界ができているとも言えます。そのサンプルプログラムがないのは良くないので、https://github.com/jsk-ros-pkg/jsk_demos/pull/1361 に作っておきました。
> eng2 モデルにエレベータのかごのモデルを入れて、現在の状態に応じてこのオブジェクトの位置を変えたりエレベータのドアを開閉するプログラムをeus上で作ってirtviewerで確認できたら、あとは、そのままpublshしてrvizでビジュアライズする、見たいにするのが、いろいろ一番簡単な作り方かと思います。Eus上でロボットの動きのアニメーションできたら、実機でそのまま動く、みたいな感覚です。
> 
> Naoto Tsukamoto, 昨日 11:31
> 岡田先生ありがとうございます．サンプルプログラム試してみました．部屋の構造や家具があるとだいぶ建物の規模感が出てくる感じがしています．
> 
> Naoto Tsukamoto, 昨日 11:34
> 引きで2号館を出してみるとこんな感じで，73B2の壁を一部消してみたいとか，床のalpha値変える・色変えてみたいとかそんな気持ちになってきます．
> 
> Koki Shinjo, 昨日 12:32
> シェーディングいじりたくなる色合いだ....
>
